var/spack/repos/builtin/packages/survey/package.py

```diff
@@ -0,0 +1,210 @@
+
+
+class Survey(CMakePackage):
+    """Survey is a high-level performance tool product from Trenza, Inc.
+       The survey collector/analytics framework is a new generation,
+       high-level, light-weight multi-platform Linux tool set that
+       targets metric collection for high-level performance analysis
+       of applications running on both single node and on large-scale
+       platforms, including the Cray platforms.
+
+       The collector is designed to work on sequential, MPI, OpenMP,
+       and hybrid codes and directly leverages several interfaces
+       available for tools inside current MPI implementations including:
+       MPICH, MVAPICH, MPT, and OpenMPI. It also supports multiple
+       architectures and has been tested on machines based on Intel,
+       AMD, ARM, and IBM P8/9 processors and integrated GPUs.
+    """
+
+    # To access the survey source and build with
+    # spack please contact: Trenza Inc. via:
+    # dmont@trenzasynergy.com
+    homepage = "http://www.trenzasynergy.com"
+    git      = "git@gitlab.com:trenza/survey.git"
+
+    maintainers = ['jgalarowicz']
+
+    version('develop', branch='master')
+    version('1.0.0', branch='1.0.0')
+
+    # MPI variants
+    variant('openmpi', default=False,
+            description="Build mpi collector for openmpi \
+                         MPI when variant is enabled.")
+    variant('mpt', default=False,
+            description="Build mpi collector for SGI \
+                         MPT MPI when variant is enabled.")
+    variant('mvapich2', default=False,
+            description="Build mpi collector for mvapich2\
+                         MPI when variant is enabled.")
+    variant('mvapich', default=False,
+            description="Build mpi collector for mvapich\
+                         MPI when variant is enabled.")
+    variant('mpich2', default=False,
+            description="Build mpi collector for mpich2\
+                         MPI when variant is enabled.")
+    variant('mpich', default=False,
+            description="Build mpi collector for mpich\
+                         MPI when variant is enabled.")
+    variant('build_type', default='RelWithDebInfo',
+            description='The build type to build',
+            values=('Debug', 'Release', 'RelWithDebInfo'))
+
+    # must have cmake at 3.12 or greater to find python3
+    depends_on('cmake@3.12:')
+
+    # for collectors
+    depends_on("libmonitor@2021.04.27+commrank", type=('build', 'link', 'run'))
+
+    depends_on("papi@5:", type=('build', 'link', 'run'))
+    depends_on("gotcha@master", type=('build', 'link', 'run'))
+    depends_on("llvm-openmp@9.0.0", type=('build', 'link', 'run'))
+
+    # MPI Installations
+    depends_on("openmpi", when='+openmpi')
+    depends_on("mpich@:", when='+mpich')
+    depends_on("mpich@2:", when='+mpich2')
+    depends_on("mvapich2@2:", when='+mvapich2')
+    depends_on("mvapich2@:1", when='+mvapich')
+    depends_on("mpt", when='+mpt')
```

Comment:
```suggestion
    depends_on("mpi", when="+mpi")
```
Then add conflicts for any MPI implementations it doesn't work with

```
conflicts("^charmpp")  # etc
```

Reply:
We need the individual depends on MPI types in order to build multiple MPI data collectors.   We don't just build one version of MPI into our tool.

Reply:
Changed to meet reviewers requested changes.